### PR TITLE
Overhaul log loading

### PR DESCRIPTION
**Pull request by CyberShadow**
Unify the channel/query log loading code.

Change algorithm to use the same filename generation algorithm as the
log-creation code.

Allow loading lines from more than one log.

Allow loading logs from all supported date formats.

Fix newline parsing being dependent on compilation target OS.

### CODE REVIEW

**Comment by un1versal** [comment]
if https://github.com/kvirc/KVIrc/issues/1710 ends up being also ~~fixed~~ implemented would be nice :)

**Comment by wodim** [comment]
If this works, you are our hero. This was hairy and buggy as fuck, and many people requested it

File: src/kvirc/ui/KviWindow.cpp [hunk 1]
```diff
@@ -1380,3 +1390,114 @@ KviIrcContext * KviWindow::context()
...
+					continue;
+
+				QList<QByteArray> list = log.split('\n');
+				unsigned int uCount = list.size();
+
+				while (uCount)
+				{
+					vLines[uCurLines++] = QString(list.at(--uCount));
+					if (uCurLines == uMaxLines)
+						goto enough;
```
Comment (un1versal) [inline src/kvirc/ui/KviWindow.cpp:1446]: goto (this reminds me of ZX-spectrum and basic programming) ahh fond memories

**Comment by CyberShadow** [comment]
OK, I think this is ready for review & testing. See the individual commits' commit messages for details on changes done.

**Comment by un1versal** [comment]
so is this ready to go? We had a few log related issues if this fix resolves any of them
see https://github.com/kvirc/KVIrc/milestone/5

Would you mind adding **fixes #issuenr** to your commit messages for any of those reports you see fixed by this, and Ill take care of the reports relabeling.

**Comment by CyberShadow** [comment]
@un1versal Added fixed issue references to commit message.

**Comment by CyberShadow** [comment]
> if #1710 ends up being also fixed implemented would be nice :)

This one is orthogonal to the fixes here.

**Comment by un1versal** [comment]
> @un1versal Added fixed issue references to commit message.

Thank you... Its preferred to close tickets automatically, but, if you really forget or are too busy to go around adding these to commit msg, if you reference them in a post here, when PR is merged, I can go and close the related issues and refer to the fix for posterity.

> This one is orthogonal to the fixes here.

Yes I see #2054 staticfocx will be pleased.

nice going, this logging thing is important for national security reasons `</sarcasm>` hahaha

**Comment by un1versal** [comment]
Im not sure if its a placebo effect or not, but when opening logviewer, it feels faster somehow.

Linux test :+1: 
Windows test: :+1: 

if none has anything else to say, Ill merge this after a short break for commercials.

**Pull request merged**